src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -4339,11 +4339,14 @@ public void visitSelect(JCFieldAccess tree) {
         }
 
         if (isType(sitesym)) {
-            if (sym.name == names._this) {
+            if (sym.name == names._this || sym.name == names._super) {
```

Comment:
I suggest to rewrite this as:

```
if (sym.name == names._this) {
   if (site.tsym == env.enclClass.sym) {
      // error
   }
} else if (sym.name == names._super) {
   if (sitesym.isInterface() || site.tsym == env.enclClass.sym) {
      // error
   }
}
```
As the compound expression is giving me headache :-)

Note that I've dropped the `constructorArgs` check, because I think this is set in Attr::visitApply, and can be assumed to be true if `isSelfCall` is set.